src/Models/AdvectedPopulations/PISCES/phytoplankton.jl

```diff
@@ -0,0 +1,377 @@
+
+    PARᴾ = get_PARᴾ(PAR¹, PAR², PAR³, bgc)
+    μᴾ = μᴵ(P, Pᶜʰˡ, PARᴾ, L_day, T, αᴾ, Lₗᵢₘᴾ, zₘₓₗ, zₑᵤ, t_darkᴾ, bgc)
+
+    println("PARᴾ = $(PARᴾ), PAR¹ = $(PAR¹), PAR² = $(PAR²), PAR³ = $(PAR³), PAR = $(PAR)")
+   # println("P = $(P), D = $(D), Z = $(Z), M = $(M), Pᶜʰˡ = $(Pᶜʰˡ), Dᶜʰˡ = $(Dᶜʰˡ), Pᶠᵉ = $(Pᶠᵉ), Dᶠᵉ = $(Dᶠᵉ), Dˢⁱ = $(Dˢⁱ), DOC = $(DOC), POC = $(POC), GOC = $(GOC), SFe = $(SFe), BFe = $(BFe), PSi = $(PSi), NO₃ = $(NO₃), NH₄ = $(NH₄), PO₄ = $(PO₄), Fe = $(Fe), Si = $(Si), CaCO₃ = $(CaCO₃), DIC = $(DIC), Alk = $(Alk), O₂ = $(O₂)")
+    #println("POC = $(POC), GOC = $(GOC), SFe = $(SFe), BFe = $(BFe)")
+  #  println("----------------------")
+  #  println("Growth terms for P are $((1-δᴾ)*μᴾ*P)")
+  #  println("Decay terms for P are $(mᴾ*K_mondo(P, Kₘ)*P), $(sh*wᴾ*P^2), $(gₚᶻ*Z), $(gₚᴹ*M), with sum $(mᴾ*K_mondo(P, Kₘ)*P + sh*wᴾ*P^2 + gₚᶻ*Z + gₚᴹ*M)")
+  #  println("Total sum for P = ", (1-δᴾ)*μᴾ*P - mᴾ*K_mondo(P, Kₘ)*P - sh*wᴾ*P^2 - gₚᶻ*Z - gₚᴹ*M)
+
+    return (1-δᴾ)*μᴾ*P - mᴾ*K_mondo(P, Kₘ)*P - sh*wᴾ*P^2 - gₚᶻ*Z - gₚᴹ*M    #eq 1
+end
+
+@inline function (bgc::PISCES)(::Val{:D}, x, y, z, t, P, D, Z, M, Pᶜʰˡ, Dᶜʰˡ, Pᶠᵉ, Dᶠᵉ, Dˢⁱ, DOC, POC, GOC, SFe, BFe, PSi, NO₃, NH₄, PO₄, Fe, Si, CaCO₃, DIC, Alk, O₂, T, PAR, PAR¹, PAR², PAR³, zₘₓₗ, zₑᵤ, Si̅, D_dust, Ω)
+    # the signature of this function is always `Val(name), x, y, z, t` and then all the tracers listed in `required_biogeochemical_tracers`, and then `required_biogeochemical_auxiliary_fields`
+    δᴰ = bgc.exudation_of_DOC.D
+    mᴰ = bgc.phytoplankton_mortality_rate.D
+    Kₘ = bgc.half_saturation_const_for_mortality
+    wᴾ = bgc.min_quadratic_mortality_of_phytoplankton
+    wₘₐₓᴰ = bgc.max_quadratic_mortality_of_diatoms
+    αᴰ = bgc.initial_slope_of_PI_curve.D
+
+    ϕ₀ = bgc.latitude
+    L_day_param = bgc.length_of_day
+    ϕ = get_ϕ(ϕ₀, y)
+    L_day = get_L_day(ϕ, t, L_day_param)
+
+    #equaitons here
+    sh = get_sh(z, zₘₓₗ)
+
+    g_Dᶻ = get_grazingᶻ(P, D, POC, T, bgc)[3]
+    g_Dᴹ = get_grazingᴹ(P, D, Z, POC, T, bgc)[3]
+ 
+    t_darkᴰ = bgc.mean_residence_time_of_phytoplankton_in_unlit_mixed_layer.D
+
+    Lₗᵢₘᴰ = Lᴰ(D, PO₄, NO₃, NH₄, Si, Dᶜʰˡ, Dᶠᵉ, Si̅, bgc)[1]
+    PARᴰ = get_PARᴰ(PAR¹, PAR², PAR³, bgc)
+
+    wᴰ = wᴾ + wₘₐₓᴰ*(1-Lₗᵢₘᴰ) #13
+    
+    μᴰ = μᴵ(D, Dᶜʰˡ, PARᴰ, L_day, T, αᴰ, Lₗᵢₘᴰ, zₘₓₗ, zₑᵤ, t_darkᴰ, bgc)
+
+   # println("Growth terms for D are $((1-δᴰ)*μᴰ*D)")
+   # println("Decay terms for D are $(mᴰ*K_mondo(D, Kₘ)*D), $(sh*wᴰ*D^2), $(g_Dᶻ*Z), $(g_Dᴹ*M), with sum $(mᴰ*K_mondo(D, Kₘ)*D + sh*wᴰ*D^2 + g_Dᶻ*Z + g_Dᴹ*M)")
+   # println("Total sum for D = ", (1-δᴰ)*μᴰ*D - mᴰ*K_mondo(D, Kₘ)*D - sh*wᴰ*D^2 - g_Dᶻ*Z - g_Dᴹ*M)
+
+    return (1-δᴰ)*μᴰ*D - mᴰ*K_mondo(D, Kₘ)*D - sh*wᴰ*D^2 - g_Dᶻ*Z - g_Dᴹ*M    #eq 9
+end
+
+@inline function (bgc::PISCES)(::Val{:Pᶜʰˡ}, x, y, z, t, P, D, Z, M, Pᶜʰˡ, Dᶜʰˡ, Pᶠᵉ, Dᶠᵉ, Dˢⁱ, DOC, POC, GOC, SFe, BFe, PSi, NO₃, NH₄, PO₄, Fe, Si, CaCO₃, DIC, Alk, O₂, T, PAR, PAR¹, PAR², PAR³, zₘₓₗ, zₑᵤ, Si̅, D_dust, Ω)
```

Comment:
@ciadht this is where the problem is, the arguments are ordered `(x, y, z, t, bgc_tracers..., light_attenuation_fields...)` so `PAR, PAR¹, PAR², PAR³` should be the final 4 arguments (in all of the function calls